osu.Game/Screens/Edit/Timing/TimingScreen.cs

```diff
@@ -144,7 +145,31 @@ private void delete()
 
             private void addNew()
             {
-                selectedGroup.Value = Beatmap.ControlPointInfo.GroupAt(clock.CurrentTime, true);
+                bool isFirstControlPoint = !Beatmap.ControlPointInfo.TimingPoints.Any();
+
+                var group = Beatmap.ControlPointInfo.GroupAt(clock.CurrentTime, true);
+
+                if (isFirstControlPoint)
+                    group.Add(new TimingControlPoint());
+                else
+                {
+                    // Try and create matching types from the currently selected control point.
+                    var selected = selectedGroup.Value;
+
+                    if (selected != null)
+                    {
+                        foreach (var controlPoint in selected.ControlPoints)
+                        {
+                            if (Activator.CreateInstance(controlPoint.GetType()) is ControlPoint copy)
+                            {
+                                copy.CopyFrom(controlPoint);
+                                group.Add(copy);
+                            }
```

Comment:
Any reason not to do

```suggestion
                            group.Add(copy.DeepClone());
```

instead?

Reply:
That's the method I was looking for. Couldn't find it immediately.